### PR TITLE
Merge Dev into Main

### LINKED ISSUE #55: Incorporate data normalization for relevant demographics
**Is your feature request related to a problem? Please describe.**
It doesn't appear that Splink handles data normalization.  There are several demographics that either benefit from, or essentially require data normalization.  For example:

* Different punctuation for patient names.  E.g. "Jean-Claude Van Damme" vs. "Jean Claude Van Damme".
* Different phonetic spellings of names.  E.g., "Jon" vs. "John".
* Different abbreviations in both names and addresses.  E.g., "St" vs. "St." vs. "Street".  Or "Jr." vs. Junior" for names.

**Describe the intended behavior**
The demographics for these two patients probably should be normalized prior to submission to the record linking process.
Jon Smith at 123 Main Street
John Smith at 123 Main St. 

**Describe the solution you'd like**
Data normalization is a pretty complicated problem.  There are different solutions for different problems.  You may be able to use a USPS process to normalize address data (though you should consider the $$ cost of calling a USPS service for large number of patients, which will be necessary for the populations we are talking about).  There may be other options to normalize phonetic spelling of names.

### PR DESCRIPTION
**Pull request by IsaacMilarky**
## Result

Merge Dev into Main to Push Recent Changes into Production

Changes:

* Upgrade [Splink to New Major Version 4.0](https://moj-analytical-services.github.io/splink/blog/2024/07/24/splink-400-released.html)
* Add Numeral Normalization Logic
* Add SBOM section to README.md

Closes #55 

## Test Plan

`make test`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ authors = ["Pat Sier <patrick.sier@cms.hhs.gov>",
 
 [tool.poetry.dependencies]
 python = "^3.10"
-splink = "^3.9.15"
+splink = "^4.0.0"
```
Comment (decause-gov) [inline pyproject.toml:10]: Major release! Nice!

**Review by decause-gov — COMMENTED** [review]
I reckon the missing docstring for the qrda check is ok since it's not implemented yet, and I can't say whether the Splink version bumps work or not, but I don't see anything obviously blocking in there, so, LGTM +1 🚢 ➿

**Comment by IsaacMilarky** [comment]
> I reckon the missing docstring for the qrda check is ok since it's not implemented yet, and I can't say whether the Splink version bumps work or not, but I don't see anything obviously blocking in there, so, LGTM +1 🚢 ➿

The tests pass on this PR and it works when I run it on test data. What is blocking for approval?

**Comment by decause-gov** [comment]
> > I reckon the missing docstring for the qrda check is ok since it's not implemented yet, and I can't say whether the Splink version bumps work or not, but I don't see anything obviously blocking in there, so, LGTM +1 🚢 ➿
> 
> The tests pass on this PR and it works when I run it on test data. What is blocking for approval?

Nothing :) #Shipit 🚢

**Review by patsier-cms — APPROVED** [review]

**Comment by decause-gov** [comment]
Nothing, ship it :ship: :)

Remy DeCausemaker
Open Source Lead
Digital Service at CMS.gov

________________________________

**Pull request merged**